plugins/workspace-search/src/workspace_search.ts

```diff
@@ -0,0 +1,581 @@
+   *     or hits enter on the button.
+   */
+  addActionBtn(btn: HTMLButtonElement, onClickFn: () => void) {
+    this.addBtnListener(btn, onClickFn);
+    this.actionDiv.appendChild(btn);
+  }
+
+  /**
+   * Creates the text input for the search bar.
+   * @returns A text input for the search bar.
+   */
+  protected createTextInput(): HTMLInputElement {
+    const textInput = document.createElement('input');
+    textInput.type = 'text';
+    textInput.setAttribute('placeholder', this.textInputPlaceholder);
+    return textInput;
+  }
+
+  /**
+   * Creates the button used to get the next block in the list.
+   * @returns The next button.
+   */
+  protected createNextBtn(): HTMLButtonElement {
+    return this.createBtn('blockly-ws-search-next-btn', 'Find next');
+  }
+
+  /**
+   * Creates the button used to get the previous block in the list.
+   * @returns The previous button.
+   */
+  protected createPreviousBtn(): HTMLButtonElement {
+    return this.createBtn('blockly-ws-search-previous-btn', 'Find previous');
+  }
+
+  /**
+   * Creates the button used for closing the search bar.
+   * @returns A button for closing the search bar.
+   */
+  protected createCloseBtn(): HTMLButtonElement {
+    return this.createBtn('blockly-ws-search-close-btn', 'Close search bar');
+  }
+
+  /**
+   * Creates a button for the workspace search bar.
+   * @param className The class name for the button.
+   * @param text The text to display to the screen reader.
+   * @returns The created button.
+   */
+  private createBtn(className: string, text: string): HTMLButtonElement {
+    // Create the button
+    const btn = document.createElement('button');
+    Blockly.utils.dom.addClass(btn, className);
+    btn.setAttribute('aria-label', text);
+    return btn;
+  }
+
+  /**
+   * Add event listener for clicking and keydown on the given button.
+   * @param btn The button to add the event listener to.
+   * @param onClickFn The function to call when the user clicks on
+   *      or hits enter on the button.
+   */
+  private addBtnListener(btn: HTMLButtonElement,
+      onClickFn: (e: Event) => void) {
+    this.addEvent(btn, 'click', this, onClickFn);
+    // TODO: Review Blockly's key handling to see if there is a way to avoid
+    //  needing to call stopPropogation().
+    this.addEvent(btn, 'keydown', this, (e: KeyboardEvent) => {
+      if (e.key === 'Enter') {
+        onClickFn(e);
+        e.preventDefault();
+      } else if (e.key === 'Escape') {
+        this.close();
+      }
+      e.stopPropagation();
+    });
+  }
+
+  /**
+   * Returns the bounding rectangle of the UI element in pixel units relative to
+   * the Blockly injection div.
+   * @returns The component’s bounding box. Null in this
+   *     case since we don't need other elements to avoid the workspace search
+   *     field.
+   */
+  getBoundingRectangle(): Blockly.utils.Rect|null {
+    return null;
+  }
+
+  /**
+   * Positions the zoom-to-fit control.
```

Comment:
can you fix this text while you're here? it's wrong in both name and description

Reply:
Done!